api-registration/server/index.js

```diff
@@ -0,0 +1,59 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  argon2.hash(password)
+    .then(hashedPassword => {
+      const sql = 'insert into "users" ("username", "hashedPassword") values ($1, $2) returning * ';
```

Comment:
This will return all of the columns from the database table, but we don't want to send the `hashedPassword` back to the client. In the `returning` statement you need to specify the columns you want returned:
```
returning "userId", "username", "createdAt"
```